src/api/c/convolve.cpp

```diff
@@ -113,9 +115,28 @@ af_err convolve2_sep(af_array *out, af_array col_filter, af_array row_filter, co
 
         af_dtype signalType  = sInfo.getType();
 
-        dim4 signalDims = sInfo.dims();
+        dim4 sdims = sInfo.dims();
+
+        ARG_ASSERT(1, (sdims.ndims()>=2));
+
+        if (cfInfo.isScalar() && rfInfo.isScalar()) {
+            af_array colArray = 0;
+            af_array rowArray = 0;
+            af_array filter = 0;
+
+            AF_CHECK(af_tile(&colArray, col_filter, sdims[0], sdims[1], sdims[2], sdims[3]));
+            AF_CHECK(af_tile(&rowArray, row_filter, sdims[0], sdims[1], sdims[2], sdims[3]));
+            AF_CHECK(af_mul (&filter, colArray, rowArray, false));
+
+            AF_CHECK(af_mul(out, signal, filter, false));
```

Comment:
I don't think its needed here, it is just too much redundancy. It is the final af_* call in that branch of code flow and `af_mul` does the swap you are talking about.